bulk_indexer.go

```diff
@@ -240,9 +255,74 @@ func (b *bulkIndexer) Flush(ctx context.Context) (BulkIndexerResponseStat, error
 	if err := jsoniter.NewDecoder(res.Body).Decode(&resp); err != nil {
 		return resp, fmt.Errorf("error decoding bulk response: %w", err)
 	}
+
+	for _, biri := range resp.FailedDocs {
+		if biri.Status == http.StatusTooManyRequests {
+			startlnIdx := biri.Position * 2
+			endlnIdx := startlnIdx + 2
+
+			if b.gzipw != nil {
+				gr, err := gzip.NewReader(bytes.NewReader(copyBuf))
+				if err != nil {
+					return resp, fmt.Errorf("failed to decompress request payload: %w", err)
+				}
+				defer gr.Close()
+
+				seen := 0
+
+				buf := make([]byte, 1024)
```

Comment:
Could we reuse this buffer after its contents are written to the `gzip.Writer`? We're creating at least a 1KB `[]byte` for each failed item. If that buffer is expanded, I imagine it will be more efficient to recycle the byte slice after writing to the gzip compressor.